projects/kit/components/input-copy/input-copy.component.ts

```diff
@@ -87,14 +87,19 @@ export class TuiInputCopyComponent
     get hintText$(): Observable<PolymorpheusContent> {
         return this.copyTexts$.pipe(
             switchMap(texts =>
-                this.copy$.pipe(
-                    switchMap(() =>
-                        merge(
+                this.copied$.pipe(
+                    switchMap(copied => {
+                        if (!copied) {
+                            return of(texts[0]);
+                        }
+
+                        this.copied$.next(false);
```

Comment:
This feels like a pretzel. Ideally it would just be a single stream without shooting itself back :) We gotta refactor that in 4.0, as for 3.0 we can probably merge this.